okta hosted sign-in widget blog post (#1561)
oktadeveloper/okta-blog: _source/_posts/2025-02-07-okta-hosted-sign-in-widget.md

---
layout: blog_post
title: "From Meh to Wow: Customize Your Okta Sign-In Experience"
author: nesh-popovic
by: contractor
communities: [javascript]
description: "Tutorial on customizing your Okta hosted sign-in widget to the next level."
tags: [sign-in-custom-domain, custom-domain, okta-hosted-sign-in]
tweets:
- ""
- ""
- ""
image: blog/okta-hosted-sign-in-widget/sign-in-widget-social-image.jpeg
type: awareness
---
The Okta Sign-In Widget is a powerful tool that allows developers to integrate Okta authentication into their web applications seamlessly. But did you know you can also customize the look and feel of the widget to match your application(s)' design, even if the widget is hosted on Okta? Whether you want to modify colors, add custom branding, or adjust the layout, Okta provides flexibility to make the widget fit your brand's identity.

In this guide, we'll transform the Okta-hosted widget using standard Javascript and CSS and its built-in functionality.

{:toc}

**Before**

> ![Giphy of default static Okta hosted sign-in widget](_source/_assets/img/blog/okta-hosted-sign-in-widget/okta-hosted-sign-in-widget-before.gif){: .center-image }

Reviewer: ```suggestion
{% img blog/okta-hosted-sign-in-widget/okta-hosted-sign-in-widget-before.gif alt:"animated gif of default static Okta hosted sign-in widgets" width:"800" %}{: .center-image }
```

Our goal:

**After**

> ![Giphy of customized Okta hosted sign-in widget with sliding window effect for new signups](_source/_assets/img/blog/okta-hosted-sign-in-widget/okta-hosted-sign-in-widget-after.gif){: .center-image }

Reviewer: ```suggestion
{% img blog/okta-hosted-sign-in-widget/okta-hosted-sign-in-widget-after.gif  alt:"animated gif of customized Okta hosted sign-in widget with sliding window effect for new signups" width:"800" %}{: .center-image }

```

---
layout: blog_post
title: "From Meh to Wow: Customize Your Okta Sign-In Experience"
author: nesh-popovic

> by: contractor

Reviewer 2: I think we have an internal-contributor category

## Okta-hosted sign-in widget basics
Before diving into customization, let's go over a few basics. "Okta-hosted widget" means that when the user needs to authenticate, you redirect them from your application to Okta, and the login screen that the user is presented with is hosted by Okta (it could be on a custom domain). Hence, all the widget code is on Okta itself (and not in the application), which makes it very convenient for your developers.

**Prequisites to this tutorial**
This guide assumes you already have a Single Sign-On (SSO) application that redirects to Okta for authentication. In addition, you already have a customized Okta-hosted widget with a [custom domain](https://help.okta.com/en-us/content/topics/settings/settings-configure-custom-url.htm) enabled to make the necessary edits I will be demonstrating in this tutorial. And lastly, this guide assumes you have [Self-Service Registration (SSR)](https://support.okta.com/help/s/article/sign-up-link-missing-from-okta-login-page-during-sp-initiated-login?language=en_US) also enabled to handle user self-registration.  

> If you don't have the above setup just yet, you can test one of our [samples](https://github.com/okta-samples) with a free developer org under (Sign up free for Developer Edition)[https://developer.okta.com/signup/]. 

Reviewer 2: ```suggestion
If you don't have the above setup just yet, you can test one of our [samples](https://github.com/okta-samples) with a free Developer Edition Account under (Sign up free for Developer Edition)[https://developer.okta.com/signup/]. 
```

>**Note:** You must use your work email address to sign up for a free developer account.

Within the samples, please refer to the ones labeled **Okta Hosted Login** or **Redirect Model**. For example, here is one in [React](https://github.com/okta-samples/okta-react-sample) and [Angular](https://github.com/okta-samples/okta-angular-sample). Each sample readme should have directions on creating an SSO application within Okta, but for more info, you can always refer to this page on [creating app integrations](https://help.okta.com/en-us/content/topics/apps/apps_app_integration_wizard_oidc.htm.). One last note, be sure you are using the correct authorization server for SSO and that is the [Okta Org Authorization Server](https://developer.okta.com/docs/concepts/auth-servers/#org-authorization-server). 

> For even more examples of getting started with the Okta Hosted Sign-in Widget, check out this blog [A Secure and Themed Sign-in Page](blog/2023/01/12/signin-custom-domain#set-up-the-angular-micro-frontend-site-and-add-okta).

Reviewer 2: ```suggestion
For even more examples of getting started with the Okta Hosted Sign-in Widget, check out this post that walks you through the basic customizations and setting up a custom domain.

{% excerpt /blog/2023/01/12/signin-custom-domain %}
```

You could consider this to break up walls of text with embedded links. 🤷🏼

## Customizing colors and branding on the sign-in widget
One of the easiest and basic ways to personalize the Okta Sign-In Widget is through the **Customizations** > **Brands** > **[your custom brand]** > **Theme** tab; here, you can modify the logo, colors, and background. 

## Understanding the sign-in widget structure
We're not here to change the button color or simply slap a logo on the widget; we're here to learn how to slice and dice that widget so it dazzles and provides a better user experience. 🙂

> We will accomplish this by utilizing the standard Javascript and CSS.

Reviewer 2: ```suggestion
We will accomplish this by utilizing standard Javascript and CSS.
```

If you use the Okta-hosted widget, you will find it under **Customizations** > **Brands** > **[your custom brand]** > **Pages** > **Sign-in** page. Click the **Configure** button and toggle the **Code editor** to **ON**. At this point, you should see some code appear like the one below:

> ```HTML

Reviewer 2: ```suggestion
```html
```

I'm surprised you don't need to wrap the html block with variables in the raw tag

           OktaUtil.completeLogin,
           function(error) {
               // Logs errors that occur when configuring the widget.
               // Remove or replace this with your own custom error handler.
               console.log(error.message, error);
           }
       );
</script>
</body>
</html>
```

Let's make a note of a few key things:
- `OktaUtil` is a global object that contains all of the widget methods and settings
- `var config = OktaUtil.getSignInWidgetConfig();` gets us all the widget configurations so we can start modifying them 
- `var oktaSignIn = new OktaSignIn(config);` initializes the widget with the config values

## Add custom CSS and third-party libraries to the sign-in widget
If you want more fine-grained control over the widget's appearance, Okta allows you to inject your own CSS styles as well as use other libraries (e.g., Bootstrap, jQuery, etc.). You can override specific elements or apply custom styles across the widget. To apply your own CSS, you can either do the inline styling or point to a CSS file (which needs to be accessible from the web). Let's dive in! 

Within the `<head>` section and below the following code,

> ```HTML

Reviewer 2: ```suggestion
```html
```

I tink you need raw wrappers here too? please add to all places with `{{ }}`

<!-- Favicon from theme -->
    <link rel="shortcut icon" href="{{faviconUrl}}" type="image/x-icon"/>
```

insert the following code that points to the Font Awesome library, which is hosted on my AWS S3 bucket (you can use your own Font Awesome or any other library, but you would need to modify the code accordingly).

> ```CSS

Reviewer 2: ```suggestion
```css
```

syntax highlighting requires lowercase

https://github.com/rouge-ruby/rouge/blob/master/docs/Languages.md
